engine/schema/src/main/java/com/cloud/domain/dao/DomainDetailsDaoImpl.java

```diff
@@ -98,7 +107,24 @@ public Scope getScope() {
 
     @Override
     public String getConfigValue(long id, ConfigKey<?> key) {
-        DomainDetailVO vo = findDetail(id, key.key());
+        DomainDetailVO vo = null;
+        String enableDomainSettingsForChildDomain = _configDao.getValue("enable.domain.settings.for.child.domain");
```

Comment:
this string should probably come from some static final as now it is encoded twice (at least)